sr_utilities_common/test/aws_manager_test.py

```diff
@@ -14,49 +14,118 @@
+        self.filename = get_filename()
 
-    def aws_get_bucket_structure_with_prefix(self):
-        result_1 = self.aws_manager.get_bucket_structure_with_prefix(BUCKET_NAME, None)
-        result_2 = self.aws_manager.get_bucket_structure_with_prefix(BUCKET_NAME, "test")
-        return result_1 is None and result_2 is not None
+    @classmethod
+    def tearDownClass(self):
+        pass
 
-    def aws_download(self):
-        self.aws_manager.download(BUCKET_NAME, rospkg.RosPack().get_path('sr_utilities_common'),
-                                  'test', ["test.txt"])
-        if os.path.exists("test.txt"):
-            os.remove("test.txt")
-            return True
-        return False
+    def test_aws_01_upload(self):
+        result = False
+        self.make_files_to_be_uploaded()
+        self.aws_manager.upload(BUCKET_NAME, "/tmp", "upload", [self.filename], None)
+        folder_list = self.aws_manager.get_bucket_structure_with_prefix(BUCKET_NAME, None)
+        if folder_list is not None:
+            for element in folder_list:
+                if element['Key'] == f"upload/{self.filename}":
+                    result = True
+        self.assertTrue(result)
 
-    def aws_upload(self):
-        self.aws_manager.upload(BUCKET_NAME, rospkg.RosPack().get_path('sr_utilities_common'),
-                                'test', ["test.txt"])
-        folder_list = self.aws_manager.get_bucket_structure_with_prefix(BUCKET_NAME, 'test')
+    def test_aws_02_upload_subfolder(self):
+        self.filename = get_filename()  # Second file should have new filename
+        result = False
+        self.make_files_to_be_uploaded()
+        self.aws_manager.upload(BUCKET_NAME, "/tmp", "upload", [self.filename], "Subfolder")
+        folder_list = self.aws_manager.get_bucket_structure_with_prefix(BUCKET_NAME, None)
         if folder_list is not None:
             for element in folder_list:
-                if element['Key'] == 'test/test.txt':
-                    return True
-        return False
+                if element['Key'] == f"Subfolder/{self.filename}":
+                    result = True
+        self.assertTrue(result)
 
+    def test_aws_03_download(self):
+        result = False
+        delete_download_folder()
+        self.aws_manager.download(BUCKET_NAME, "/tmp", "download", [self.filename], "upload")
+        if os.path.exists(f"{DOWNLOAD_PATH}/{self.filename}"):
+            result = True
+        self.assertTrue(result)
 
-if __name__ == '__main__':
-    rospy.init_node('aws_manager_test')
+    def test_aws_04_download_subfolder(self):
+        result = False
+        self.aws_manager.download(BUCKET_NAME, "/tmp", "download", [self.filename], "Subfolder")
+        if os.path.exists(f"{DOWNLOAD_PATH}/{self.filename}"):
+            result = True
+        self.assertTrue(result)
+
+    def test_aws_05_remove_all_files(self):
+        rospy.sleep(4)
```

Comment:
why this sleep?

Reply:
Because of the delay of uploading files to AWS, the rostest cant account for this, so I put the sleep here to ensure all files are uploaded before removing everything, the test in total takes about 10 seconds so its a marginal change